app/assets/javascripts/components/sidebar/posts_main_component.js.coffee

```diff
@@ -0,0 +1,44 @@
+# @state.posts is an array of json objects describing posts
+window.PostsMainComponent = React.createClass
+	getInitialState: ->
+		{posts_json: []}
+
+	componentWillMount: ->
+		@loadPostsFromServer()
+	
+	componentDidMount:->
+		@pollServer()	
+
+	loadPostsFromServer: ->
+		console.log "loading posts from server"
+		url = 'posts' + (if @props.postsAreRelevant then '/relevant' else '')
+		return $.ajax
+			url: url+ '.json'
+			dataType: 'json'
+			cache: false
+			success: (posts) =>
+				@setState {posts_json: posts}
+
+			error: (jqXHR, textStatus, errorThrown) ->
+				console.log "Error updating posts: "+errorThrown
+
+
+	pollServer: ->
+		setInterval(@loadPostsFromServer, @props.pollInterval)
+
+	render: ->
+		postClass = if @props.postsAreRelevant then "relevant_posts" else "all_posts"
```

Comment:
why not have the parent just pass in the className of the child, since all you're doing off that boolean is changing a class, why not just pass it directly?